### ISSUE TITLE
Kernel crash using default_value_for

### ISSUE THREAD

**Issue by yves-surrel** [enthought/traitsui#313]
The following program (if the last line is uncommented) causes a kernel crash

     from traitsui.api import View
     
     v = View('a')
     
     vc = v.content
     
     # With **layout** trait of v.content, all is fine:
     print(vc.layout)
     # output: 'normal'
     layout_trait = vc.trait('layout')
     print(layout_trait.default_value_for(vc,'layout'))
     # output: 'normal'
     
     # The same with **height** trait of v.content
     print vc.height
     # output: -1.0
     height_trait = vc.trait('height')
     # uncomment to crash:
     #print(height_trait.default_value_for(vc,'height'))

With both wx and Qt backends, and on windows and MacOSX platforms

**Comment by jdmarch** [comment]
@yves-surrel please edit to use markdown for code blocks. Thanks.

**Comment by corranwebster** [comment]
I can replicate.  Minimal example:
```
from traits.api import HasTraits, Float, Property


class Test(HasTraits):

    x = Float

    y = Property(Float, depends_on='x')

    def _get_y(self):
        return self.x+1

if __name__ == '__main__':
    t = Test()
    print(t.y)
    y_trait = t.trait('y')
    
    # This crashes Python
    y_trait.get_default_value_for(t, 'y')
```
so this is a Traits issue, not a TraitsUI issue.  I will open a ticket there.

That said, `default_value_for` in this case not clear what it should mean, since this is a `Property` trait (on the `contents` which are a `Group`), and so doesn't really have a default value, rather it only has whatever the current value should be based upon the dependencies it has.

So I think the immediate fix is "don't do that" (which is unsatisfying, I realize).

I will close this after I have opened the ticket in Traits.

**Comment by mdickinson** [comment]
FTR, this is now fixed in traits: enthought/traits#350.
